Email Service/EmailService.cls

```diff
@@ -0,0 +1,72 @@
+global class EmailService implements Messaging.InboundEmailHandler {
+    global Messaging.InboundEmailResult handleInboundEmail(Messaging.inboundEmail email, Messaging.InboundEnvelope env){
+        Messaging.InboundEmailResult result = new Messaging.InboundEmailResult();
+        try{
+            if(email.textAttachments.size() > 0 && email.textAttachments[0].filename.endsWith('.xml'))
+            {
+                String myPlainText= '';
+                myPlainText = email.plainTextBody;
+                List<Contact> contactList = new List<Contact>();
+                if(email.textAttachments != null)
```

Comment:
is this sensible? 
on line 5 we are checking email.textAttachments.size() and email.textAttachments[0].filename.endsWith('.xml')
I don't think email.textAttachments will be null in any case here

Reply:
yes, I changed it.